Homework3/Novikov_homework3.cpp

```diff
@@ -0,0 +1,261 @@
+  std::cin >> finder;
+  if (arraySize > 0) {
+    for (size_t i = 0; i < arraySize; i++) {
+      if (array[i] == finder) {
+        deleteElement(array, i, arraySize);
+      }
+    }
+  } else
+    std::cout << "No elements left\n";
+}
+
+void TaskThreeDeleteElement() {
+  constexpr int maxArraySize = 10;
+  int ArraySize = maxArraySize;
+  int MyArray[maxArraySize];
+  int maxRange{};
+  int minRange{};
+  std::cout << "Enter element max range\n";
+  std::cin >> maxRange;
+  std::cout << "Enter element min range\n";
+  std::cin >> minRange;
+
+  std::mt19937 rnd;
+  std::uniform_int_distribution<> rand_distribution(maxRange, minRange);
+  for (size_t i = 0; i < maxArraySize; i++) {
+    MyArray[i] = rand_distribution(rnd);
+  }
+
+  std::cout << "Generated array: [ ";
+  outputArray(MyArray, ArraySize);
+  std::cout << "]\n";
+
+  enum oportunity { exit = 0, deleteElement };
+  int choise{};
+  bool menu{true};
+  while (menu) {
+    std::cout << "Choose what to do: \n";
+    std::cout << "1. Delete element \n";
+    std::cout << "0. Exit \n";
+    std::cin >> choise;
+    switch (choise) {
+    case exit:
+      menu = false;
+      break;
+    case deleteElement:
+      findElement(MyArray, ArraySize);
+      break;
+    }
+  }
+  std::cout << "Result array: [ ";
+  outputArray(MyArray, ArraySize);
+  std::cout << "]\n";
+}
+
+void TaskFourToUpper() {
+  // clear cin buffer after last input
+  std::cin.clear();
+  std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+
+  constexpr int kMaxStringSize{255};
+  char string[kMaxStringSize]{};
+  std::cout << "Enter string less than " << kMaxStringSize << " symbol\n";
+
+  while (true) {
+    if (!std::cin.getline(string, kMaxStringSize)) {
+      std::cout << "Error in string range, try again\n";
+      std::cin.clear();
+      std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+    } else if (!strlen(string))
+      std::cout << "Empty string try again\n";
+    else
+      break;
+  }
+
+  int i{0};
+  while (*(string + i)) {
+    if (*(string + i) >= 'a' && *(string + i) <= 'z') {
+      *(string + i) -= 32;
+    }
+    i++;
+  }
+  std::cout << "Result string: " << string << std::endl;
+}
+
+void TaskFiveisDigit() {
+  // clear cin buffer after last input
+  std::cin.clear();
+  std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+
+  constexpr int kMaxStringSize{255};
+  char string[kMaxStringSize]{};
+  std::cout << "Enter string less than " << kMaxStringSize << " symbol\n";
+
+  while (true) {
+    if (!std::cin.getline(string, kMaxStringSize)) {
+      std::cout << "Error in string range, try again\n";
+      std::cin.clear();
+      std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+    } else if (!strlen(string))
+      std::cout << "Empty string try again\n";
+    else
+      break;
+  }
+  const int cycleDuration = strlen(string);
+  std::cout << "Digits in string [ ";
+  for (size_t i = 0; i < cycleDuration; i++) {
```

Comment:
```suggestion
  for (size_t i = 0; string[i] != '\0'; i++) {
```
для этого же есть нультерминальный символ, зачем лишний раз писать strlen